scripts/generateMetadata.js

```diff
@@ -46,32 +53,55 @@ async function docgen(filePath) {
   }
 }
 
-const getComponentNameFromFile = (file) => file.replace(/^.*[\\/]/, '').replace(/\.js/, '');
+const getComponentNameFromFile = (file) => path.basename(file).replace(/\.js/, '');
 
-// Get files recursively as an array
-const getFilesFromDirectory = (directoryPath) => {
-  const filesInDirectory = fs.readdirSync(path.join(root, directoryPath));
-  const files = filesInDirectory.map((file) => {
-    const filePath = path.join(directoryPath, file);
-    const stats = fs.statSync(path.join(root, filePath));
+/** Analyzes subcomponents of a component in given path and extracts paths they're imported from. */
+function getSubcomponentPaths(componentPath) {
+  const componentName = getComponentNameFromFile(componentPath);
 
-    if (stats.isDirectory()) {
-      return getFilesFromDirectory(filePath);
-    }
+  /** Matches lines that are like `Component.Subcomponent = Subcomponent` */
+  const subcomponentRegExp = new RegExp(
+    `(${componentName}\\.[A-Z]\\w+) = (?<subcomponentName>\\w+)`,
+    'g',
+  );
 
-    return filePath;
-  });
+  const fileContent = fs.readFileSync(path.join(root, componentPath), 'utf-8');
+  const subcomponentNameMatches = fileContent.matchAll(subcomponentRegExp);
+  const subcomponentNames = [...subcomponentNameMatches].map((a) => a.groups.subcomponentName);
+
+  if (!subcomponentNames.length) return [];
+
+  /** Matches lines that are like `Subcomponent from './path/to/subcomponent';` */
+  const subcomponentPathRegExp = new RegExp(
+    `(${subcomponentNames.join('|')}) from '(?<path>[.\\w\\/-]+)';`,
+    'g',
+  );
+
+  const subcomponentPathMatches = fileContent.matchAll(subcomponentPathRegExp);
+  const subcomponentPaths = [...subcomponentPathMatches].map((match) =>
+    path.join(componentPath, '../', match.groups.path),
+  );
 
-  return files.filter((file) => file.length).flat();
-};
+  return subcomponentPaths;
+}
+
+/**
+ * Gets components that are imported in `index.js` of given directory (assuming they're all exported).
+ */
+function getExposedFilesFromDirectory(directoryPath) {
+  const indexFile = fs.readFileSync(path.join(root, directoryPath, 'index.js'), 'utf-8');
+  const importMatches = indexFile.matchAll(/from '(?<path>.+)'/g);
+  const filePaths = [...importMatches].map((match) => path.join(directoryPath, match.groups.path));
+  const subcomponentPaths = filePaths.map(getSubcomponentPaths).flat();
```

Comment:
Just wondering, not a requirement, but does this solve the issue of let's say we have Button, and button imports from '../SharedButtonProps.js', this can output `SharedButtonProps`as part of Button?

 That was a problem we had in the past.

Reply:
Do you mean, can it generate data for inherited props?

Reply:
If that's what you meant, then no it doesn't I think.

Reply:
Yeah!

Reply:
Aw man :(

Reply:
I think we can address that in another PR. It would truly be great to have that feature.